compiler/passes/convert-uast.cpp

```diff
@@ -4400,13 +4405,28 @@ bool LoopAttributeInfo::insertBlockSizeCall(Converter& converter, BlockStmt* bod
   return false;
 }
 
+bool LoopAttributeInfo::insertItersPerThreadCall(Converter& converter, BlockStmt* body) {
+  static int counter = 0;   // see comment in insertBlockSizeCall()
+
+  if (itersPerThreadAttr) {
+    auto newCall = new CallExpr("chpl__gpuItersPerThreadAttr", new_IntSymbol(counter++));
+    for (auto actual : itersPerThreadAttr->actuals()) {
+      newCall->insertAtTail(converter.convertAST(actual));
+    }
+    body->insertAtTail(newCall);
+    return true;
+  }
+  return false;
+}
```

Comment:
Since this is also copied verbatim (and since the approach might be re-used in other attributes) I think we should create two helpers:

1. One that converts an attribute into a call expression by iterating over the actuals
2. One more helper that does what 1 does but also inserts an additional counter-based argument (counter can be passed from the creator of the attribute).

That way, these functions (`insertGpuEligibilityAssertion`, `insertItersPerThreadCall`, etc.) can all have 1-line implementations

Reply:
Sure. How about put everything in just one helper and share the counter?

Reply:
Sounds good, I see no problem with sharing the counter.